### PR TITLE
New Rest service to append a set of existing modifications to a target node

### PR DESCRIPTION
**Pull request by dbraquart**

### CODE REVIEW

**Review by SlimaneAmar — CHANGES_REQUESTED** [review]

File: src/main/java/org/gridsuite/study/server/StudyService.java [hunk 1]
```diff
@@ -2223,6 +2223,20 @@ public void reorderModification(UUID studyUuid, UUID nodeUuid, UUID modification
         }
     }
 
+    @Transactional(readOnly = true)
```
[the remark this answers is not in the document]
Comment (dbraquart) [reply src/main/java/org/gridsuite/study/server/StudyService.java]: yes, this is not a read-only method!

File: src/main/java/org/gridsuite/study/server/NetworkModificationService.java [hunk 2]
```diff
@@ -443,6 +443,25 @@ public void reorderModification(UUID groupUuid, UUID modificationUuid, UUID befo
...
+        Objects.requireNonNull(sourceGroupUuid);
+        var path = UriComponentsBuilder.fromPath(GROUP_PATH
+                        + DELIMITER + MODIFICATIONS_PATH + DELIMITER + "duplicate")
+                .queryParam("sourceGroupUuid", sourceGroupUuid);
+
+        HttpEntity<String> httpEntity;
+        try {
+            HttpHeaders headers = new HttpHeaders();
+            headers.setContentType(MediaType.APPLICATION_JSON);
+            httpEntity = new HttpEntity<>(objectMapper.writeValueAsString(modificationUuidList), headers);
```
[the remark this answers is not in the document]
Comment (dbraquart) [reply src/main/java/org/gridsuite/study/server/NetworkModificationService.java:443]: I tried without, but I have to use it to convert List<UUID> into String

File: src/main/java/org/gridsuite/study/server/NetworkModificationService.java [hunk 3]
```diff
@@ -398,19 +398,30 @@ public void stopBuild(@NonNull UUID studyUuid, @NonNull UUID nodeUuid) {
...
+    }
+
+    public void reorderModification(UUID groupUuid, List<UUID> modificationUuidList, UUID beforeUuid) {
         Objects.requireNonNull(groupUuid);
-        Objects.requireNonNull(modificationUuid);
-        var path = UriComponentsBuilder.fromPath(GROUP_PATH
-                + DELIMITER + MODIFICATIONS_PATH + DELIMITER + "move")
-            .queryParam("modificationsToMove", modificationUuid);
+        var path = UriComponentsBuilder.fromPath(GROUP_PATH)
+            .queryParam("action", "MOVE");
```
Comment (Meklo) [inline src/main/java/org/gridsuite/study/server/NetworkModificationService.java:416]: Why not implement GroupModificationAction within the study server too ?

Comment (dbraquart) [reply src/main/java/org/gridsuite/study/server/NetworkModificationService.java:416]: Good point. I will decide with @SlimaneAmar if we change them.

Comment (dbraquart) [reply src/main/java/org/gridsuite/study/server/NetworkModificationService.java:416]: We wont change study server cause the logic is a bit different

**Review by Meklo — APPROVED** [review]

**Review by SlimaneAmar — APPROVED** [review]

File: src/main/java/org/gridsuite/study/server/service/StudyService.java [hunk 4]
```diff
@@ -1305,13 +1305,29 @@ public void reorderModification(UUID studyUuid, UUID nodeUuid, UUID modification
...
-            networkModificationService.reorderModification(groupUuid, modificationUuid, beforeUuid);
+            List<UUID> modificationUuidList = Collections.singletonList(modificationUuid);
+            networkModificationService.reorderModification(groupUuid, modificationUuidList, beforeUuid);
             updateStatuses(studyUuid, nodeUuid, false);
         } finally {
             notificationService.emitEndModificationEquipmentNotification(studyUuid, nodeUuid);
         }
     }
 
+    @Transactional()
```
[the remark this answers is not in the document]
Comment (dbraquart) [reply src/main/java/org/gridsuite/study/server/service/StudyService.java:1316]: yes

**Pull request merged**